### PR TITLE
FISH-786 FISH-1521 FISh-385 Revert of external security connectors integration

### PR DESCRIPTION
**Pull request by OndroMih**
After replacing the security connectors implementation with integration of external security connectors, there are failures in Jakarta EE TCK.

Reverting the following PRs related or dependent on the integration of external security connectors:

* https://github.com/payara/Payara/pull/5265
* https://github.com/payara/Payara/pull/5338

This also reverts a few other PRs that only increased the version of the connectors dependency as the new versions of connectors were released.

Leaving the update of the MicroProfile dependency from version 4.0 to 4.0.1, because that doesn't break anything and fixes a potential issue with dependencies.

### CODE REVIEW

**Review by Pandrex247 — APPROVED** [review]

**Comment by Pandrex247** [comment]
Jenkins test please

**Comment by Pandrex247** [comment]
`oauth2-client-integration:5.2021.5-SNAPSHOT`
Presumably you need to change some snapshots to finals

File: appserver/payara-appserver-modules/security-oauth2/pom.xml [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  and therefore, elected the GPL Version 2 license, then the option applies
+  only if the new code is made subject to such option by the copyright
+  holder.
+ -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>fish.payara.server.internal.payara-appserver-modules</groupId>
+        <artifactId>payara-appserver-modules</artifactId>
+        <version>5.2021.5-SNAPSHOT</version>
```
Comment (cubastanley) [inline appserver/payara-appserver-modules/security-oauth2/pom.xml:46]:
```suggestion
        <version>5.2021.5</version>
```

File: appserver/payara-appserver-modules/security-openid/pom.xml [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+ only if the new code is made subject to such option by the copyright
+ holder.
+-->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>fish.payara.server.internal.payara-appserver-modules</groupId>
+        <artifactId>payara-appserver-modules</artifactId>
+        <version>5.2021.5-SNAPSHOT</version>
```
Comment (cubastanley) [inline appserver/payara-appserver-modules/security-openid/pom.xml:47]:
```suggestion
        <version>5.2021.5</version>
```

**Review by cubastanley — CHANGES_REQUESTED** [review]
Final versions

**Comment by cubastanley** [comment]
Jenkins test please

**Comment by Pandrex247** [comment]
Force merging since MP TCKs are now attempting to run additional tests which would fail against this branch (through no fault of its own)

**Pull request merged**